src/module/functions/local_function/mod.rs

```diff
@@ -116,22 +115,34 @@ impl LocalFunction {
 
     /// Get the block associated with the given id.
     pub fn block(&self, block: BlockId) -> &Block {
-        self.exprs[block.into()].unwrap_block()
+        self.get(block.into()).unwrap_block()
     }
 
     /// Get the block associated with the given id.
     pub fn block_mut(&mut self, block: BlockId) -> &mut Block {
-        self.exprs[block.into()].unwrap_block_mut()
+        self.get_mut(block.into()).unwrap_block_mut()
     }
 
     /// Get the expression associated with the given id
     pub fn get(&self, id: ExprId) -> &Expr {
-        &self.exprs[id]
+        &self.exprs.arena[id]
     }
 
     /// Get the expression associated with the given id
     pub fn get_mut(&mut self, id: ExprId) -> &mut Expr {
-        &mut self.exprs[id]
+        &mut self.exprs.arena[id]
+    }
+
+    /// Get access to a `FunctionBuilder` to continue adding expressions to
+    /// this function.
+    pub fn builder(&self) -> &FunctionBuilder {
```

Comment:
One can't use a shared ref to a function builder to add expressions though, right?

Reply:
True! I figured it'd be good to expose in terms of idioms although we can always remove it later